tests/unit/stats/test_stats_exporters.py

```diff
@@ -0,0 +1,144 @@
+from mock import mock
+import pytest
+
+from flask import current_app
+from invenio_cache import current_cache
+
+from stats_helpers import create_stats_fixtures
+from zenodo.modules.stats.exporters import PiwikExporter,\
+    PiwikExportRequestError
+
+
+class MockResponse:
+    def __init__(self, json_data, status_code):
+        self.json_data = json_data
+        self.status_code = status_code
+        self.ok = status_code == 200
+
+    def json(self):
+        return self.json_data
+
+
+def mocked_requests_success(*args, **kwargs):
+    url = current_app.config['ZENODO_STATS_PIWIK_EXPORTER'].get('url', None)
+    json = {'status': 'success', 'invalid': 0}
+
+    if args[0] == url:
+        return MockResponse(json, 200)
+
+    return MockResponse(None, 404)
+
+
+def mocked_requests_invalid(*args, **kwargs):
+    url = current_app.config['ZENODO_STATS_PIWIK_EXPORTER'].get('url', None)
+    json = {'status': 'success', 'invalid': 1}
+
+    if args[0] == url:
+        return MockResponse(json, 200)
+
+    return MockResponse(None, 404)
+
+
+def mocked_requests_fail(*args, **kwargs):
+    url = current_app.config['ZENODO_STATS_PIWIK_EXPORTER'].get('url', None)
+
+    if args[0] == url:
+        return MockResponse({}, 500)
+
+    return MockResponse(None, 404)
+
+
+@mock.patch('zenodo.modules.stats.exporters.requests.post',
+            side_effect=mocked_requests_success)
+def test_piwik_exporter(app, db, es, locations, event_queues, full_record):
+    records = create_stats_fixtures(
+        metadata=full_record, n_records=1, n_versions=1, n_files=1,
+        event_data={'user_id': '1', 'country': 'CH'},
+        # 4 event timestamps
+        start_date=datetime(2018, 1, 1, 13),
+        end_date=datetime(2018, 1, 1, 15),
+        interval=timedelta(minutes=30),
+        do_process_events=True)
+
+    current_cache.delete('piwik_export:bookmark')
+    bookmark = current_cache.get('piwik_export:bookmark')
+    assert bookmark is None
+
+    start_date = datetime(2018, 1, 1, 12)
+    end_date = datetime(2018, 1, 1, 14)
+    PiwikExporter().run(start_date=start_date, end_date=end_date)
+    bookmark = current_cache.get('piwik_export:bookmark')
+    assert bookmark == u'2018-01-01T14:00:00'
+
+    PiwikExporter().run()
+    bookmark = current_cache.get('piwik_export:bookmark')
+    assert bookmark == u'2018-01-01T14:30:00'
+
+
+@mock.patch('zenodo.modules.stats.exporters.requests.post',
+            side_effect=mocked_requests_invalid)
+def test_piwik_exporter_invalid_request(app, db, es, locations, event_queues,
+                                        full_record):
+    records = create_stats_fixtures(
+        metadata=full_record, n_records=1, n_versions=1, n_files=1,
+        event_data={'user_id': '1', 'country': 'CH'},
+        # 4 event timestamps
+        start_date=datetime(2018, 1, 1, 13),
+        end_date=datetime(2018, 1, 1, 15),
+        interval=timedelta(minutes=30),
+        do_process_events=True)
```

Comment:
You can skip the rest of the fixture creation steps by passing `do_aggregate_events=False` and `do_update_record_statistics=False`, to save on some test duration